WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -66,51 +66,54 @@ private SystemAwakeChecker(CoinJoinManager coinJoinManager, Func<Task<IPowerSavi
 
 		protected override async Task ActionAsync(CancellationToken cancel)
 		{
-			IPowerSavingInhibitorTask? task = _powerSavingTask;
-
 			switch (CoinJoinManager.HighestCoinJoinClientState)
 			{
 				case CoinJoinClientState.Idle:
-					if (task is not null)
-					{
-						Logger.LogWarning("Computer idle state is allowed again.");
-						await task.StopAsync().ConfigureAwait(false);
-						_powerSavingTask = null;
-					}
+					await ReleaseAllPreventionAsync().ConfigureAwait(false);
+					break;
 
+				case CoinJoinClientState.InProgress or CoinJoinClientState.InCriticalPhase:
+					await PreventSleepAsync().ConfigureAwait(false);
 					break;
 
-				case CoinJoinClientState.InProgress:
-					if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-					{
-						if (task is not null)
-						{
-							if (!task.Prolong(Timeout.Add(TimeSpan.FromMinutes(1))))
-							{
-								Logger.LogTrace("Failed to prolong the power saving task.");
-								task = null;
-							}
-						}
-
-						if (task is null)
-						{
-							Logger.LogTrace("Create new power saving prevention task.");
-							_powerSavingTask = await TaskFactory!().ConfigureAwait(false);
-						}
-					}
-					else
-					{
-						await EnvironmentHelpers.ProlongSystemAwakeAsync().ConfigureAwait(false);
-					}
+				default:
+					throw new ArgumentOutOfRangeException();
+			}
+		}
 
-					break;
+		private async Task PreventSleepAsync()
+		{
+			IPowerSavingInhibitorTask? task = _powerSavingTask;
 
-				case CoinJoinClientState.InCriticalPhase:
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				if (task is not null)
+				{
+					if (!task.Prolong(Timeout))
```

Comment:
What is the reasoning for this change?

Reply:
We had this before: `Timeout.Add(TimeSpan.FromMinutes(1))`.
`Timeout` is already 5 minutes, feels redundant to make it 6 mins.

Reply:
Well, you are right but the reasoning is slightly different: 

Originally, `PeriodicRunner` had period of 5 minutes, so we needed that `+1` minute to be safe that prolonging will succeed.

Now, we have interval of 5 seconds (L22) but we prlong by 5 minutes which seems too much. Given that we check every 5 seconds, we should prolong by about 1 minute or so. If we prolong by 5 minutes, then when WW crashes, then user computer is for 5 minutes in an invalid state (can go to sleep but is prevented). Even though, even that should not happen as we guard against this on Linux but fundamentally, there may be new suported platforms so 1 minute sounds good to me. 5 sounds too much but it's a nit.

Reply:
You are completely right. Let's go with `Timeout = 1 minute` for now.